Library/Homebrew/os/mac/sdk.rb

```diff
@@ -25,14 +25,44 @@ class BaseSDKLocator
       class NoSDKError < StandardError; end
 
       def sdk_for(v)
-        path = sdk_paths[v]
-        raise NoSDKError if path.nil?
+        sdk = all_sdks.find { |s| s.version == v }
+        raise NoSDKError if sdk.nil?
 
-        SDK.new v, path, source
+        sdk
       end
 
       def all_sdks
-        sdk_paths.map { |v, p| SDK.new v, p, source }
+        @all_sdks ||= begin
+          # Bail out if there is no SDK prefix at all
+          if File.directory? sdk_prefix
```

Comment:
```suggestion
          # Bail out if there is no SDK prefix at all
          next unless File.directory? sdk_prefix
```

Reply:
That's not valid syntax for `begin ... end` assignment blocks.

Reply:
I think this is coming in a newer Ruby 😭.

In that case I'd suggest something like `return @all_sdks ||= [] unless File.directory? sdk_prefix` to avoid the whole thing below being needing nested.

Reply:
Slight catch with that is the `unless File.directory? sdk_prefix` won't be cached. Does it matter much?

Reply:
Would be if you do `@all_sdks ||= []`, no?

Reply:
Isn't `return @all_sdks ||= [] unless File.directory? sdk_prefix` functionally equivalent to:

```
unless File.directory? sdk_prefix
  return @all_sdks ||= []
end
```

?

Reply:
A fully cached solution would probably be:

```
def all_sdks
  return @all_sdks if @all_sdks

  @all_sdks = []

  return @all_sdks unless File.directory? sdk_prefix

  # Populate @all_sdks

  @all_sdks
end
```

Reply:
Ah, sorry, I see your point here.  Yes, your solution looks good.